### PR TITLE
New jsonop

### PR DESCRIPTION
**Pull request by harishkumarv**

### CODE REVIEW

File: lib/Cache.js [hunk 1]
```diff
@@ -125,25 +128,27 @@ class Cache {
...
 						continue keyloop; //eslint-disable-line
 					}
 				}
-				if (this.knowledge[key].contains(value)) {
-					keyValues[key] = { value, item: entity };
-				}
 
-				if (slice.link) { for (const linkType in slice.link) {
-					const dKey = this.getDualKey(slice, linkType);
+				keyValues[key] = { value, item: entity };
```
Comment (aravindet) [inline lib/Cache.js:139]: Why remove the `if (this.knowledge[key].contains(value))` check?

Comment (harishkumarv) [inline lib/Cache.js:139]: If the current knowledge is say [ [1, 5] ] and we add an item with order value 7?
or if we update the order property of  an item from 3 to 6.

The value won't be in the range. So we would want to add a new range and change it to something like [ [1, 5], [6] ].

Comment (aravindet) [inline lib/Cache.js:139]: Correct, where are you creating this new point range ([6, 6])?

Without doing that, I would prefer to skip adding this item altogether, I can't see any harm in doing that.

Reviewed, everything else is good.

File: tests/cache-entity.test.js [hunk 2]
```diff
@@ -33,16 +29,14 @@ test.cb('checking if callback of getEntity is fired if entity doesnt exist', t =
 			cache.put({ entities: { alice: null } });
 		}
 	});
-
 	t.deepEqual(
 		cache.getEntity('alice', (err, entity) => {
-			t.truthy(!err, 'threw an error');
-			t.truthy(!entity, 'entity doesnt exist');
+			t.true(!err, 'threw an error');
```
Comment (satya164) [inline tests/cache-entity.test.js]: we should just change these to `t.falsy(err, ...)`.

File: tests/infibug.test.js [hunk 3]
```diff
@@ -46,6 +43,7 @@ test('should avoid duplicates', t => {
 		] }
 	});
 
+	console.log(cache.indexes.arr);
```
Comment (satya164) [inline tests/infibug.test.js]: nit: extra console log

File: lib/Cache.js (hunk 1 above)
Comment (harishkumarv) [inline lib/Cache.js:139]: It is being handled in the putEntity function.
We delete the original item and add the new item.
The knowledge generated by the result of changing order property from a value in range to something out of range has two ranges.

Example: 
this.knowledge for a key is say [ [1, 6] ] and I change the order property from 3 to 7.

allChanges now has the a range array that looks like [ [3,3], [7,7] ].

**Pull request merged**